src/common/rect.h

```diff
@@ -0,0 +1,46 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  A simple rectangle helper class.
+ */
+
+#ifndef COMMON_RECT_H
+#define COMMON_RECT_H
+
+namespace Common {
+
+class Rect {
+public:
+	Rect();
+
+	operator bool() const;
+	bool operator!() const;
+
+	bool operator==(const Rect &rect) const;
+	bool operator!=(const Rect &rect) const;
+
+	double x, y;
+	double w, h;
```

Comment:
You sure you want to work with floating point values here, instead of integers like SDL_Rect?

Keep in mind that absolute equality of two such rectangles (i.e. `operator==()`) that have been created by different calculations might be difficult. You might need to substract the coordinates and compare to an epsilon instead.

Not saying that you should do that *now*, just something to keep in mind if you really want to keep the doubles and depend on absolute equality.

Reply:
Or, at the very least, make the underlying type a template parameter.